### ISSUE TITLE
Return type hint format not enforced

### ISSUE THREAD

**Issue by mbomb007** [slevomat/coding-standard#1403]
Here is the code piece of code to format.

**Test.php**

```php
<?php

namespace Drupal\u_gpa;

/**
 * Represents a test.
 */
class Test {

  /**
   * Attempts to get a connection to the database and run the query.
   *
   * @return array | null
   *   An array of result set rows, or NULL
   */
  public function executeQuery(): array | NULL {
    // Other code would be here...
    return NULL;
  }

}

```

Notice that the return type in the comment is formatted incorrectly, and outside too. According to the rules I've set below, return types should have no spaces, use short nullable, and null should be last. So it should be `array|null` in the comment, and `?array` outside.

The relevant rules I'm using:

**custom_ruleset.xml**

```xml
<?xml version="1.0"?>
<ruleset name="Custom">
  <rule ref="SlevomatCodingStandard.TypeHints.NullableTypeForNullDefaultValue" />
  <rule ref="SlevomatCodingStandard.TypeHints.ParameterTypeHint">
    <exclude name="SlevomatCodingStandard.TypeHints.ParameterTypeHint.UselessAnnotation" />
    <exclude name="SlevomatCodingStandard.TypeHints.ParameterTypeHint.MissingTraversableTypeHintSpecification" />
  </rule>
  <rule ref="SlevomatCodingStandard.TypeHints.PropertyTypeHint">
    <exclude name="SlevomatCodingStandard.TypeHints.PropertyTypeHint.UselessAnnotation" />
    <exclude name="SlevomatCodingStandard.TypeHints.PropertyTypeHint.MissingTraversableTypeHintSpecification" />
  </rule>
  <rule ref="SlevomatCodingStandard.TypeHints.ReturnTypeHint">
    <exclude name="SlevomatCodingStandard.TypeHints.ReturnTypeHint.UselessAnnotation" />
    <exclude name="SlevomatCodingStandard.TypeHints.ReturnTypeHint.MissingTraversableTypeHintSpecification" />
  </rule>
  <rule ref="SlevomatCodingStandard.TypeHints.ReturnTypeHintSpacing">
    <property name="spacesCountBeforeColon" value="1" />
  </rule>
  <rule ref="SlevomatCodingStandard.TypeHints.UnionTypeHintFormat">
    <property name="withSpaces" value="no" />
    <property name="shortNullable" value="yes" />
    <property name="nullPosition" value="last" />
  </rule>
</ruleset>
```

With the incorrect casing and spacing of `array | NULL`, and not using `?array`, it doesn't have any sniff showing anything.

Test with the following command:

```bash
vendor/bin/phpcs -ps Test.php --standard=custom_ruleset.xml --runtime-set testVersion 8.1
```

**Comment by kukulich** [comment]
I'm not able to reproduce the bug. Please try to provide failing test.
